packages/abtest/src/index.ts

```diff
@@ -0,0 +1,48 @@
+/**
+ * External dependencies
+ */
+import { createExPlatClient } from '@automattic/explat-client';
+import createExPlatClientReactHelpers from '@automattic/explat-client-react-helpers';
+
+/**
+ * Internal dependencies
+ */
+import { isDevelopmentMode } from './utils';
+import { logError } from './error';
+import { fetchExperimentAssignment } from './assignment';
+import { getAnonId } from './anon';
+
+declare global {
+	interface Window {
+		wcTracks: {
+			isEnabled: boolean;
+		};
+	}
+}
+
+const exPlatClient = createExPlatClient( {
+	fetchExperimentAssignment,
+	getAnonId,
+	logError,
+	isDevelopmentMode,
+} );
+
+export const {
+	loadExperimentAssignment,
+	dangerouslyGetExperimentAssignment,
+} = exPlatClient;
+const exPlatClientReactHelpers = createExPlatClientReactHelpers( exPlatClient );
+const { Experiment: ExperimentOriginal } = exPlatClientReactHelpers;
+
+export const Experiment = ( props: {
+	name: string;
+	defaultExperience: React.ReactNode;
+	treatmentExperience: React.ReactNode;
+	loadingExperience: React.ReactNode;
+} ): JSX.Element | React.ReactNode => {
+	if ( window.wcTracks.isEnabled ) {
```

Comment:
Instead of this I recommend altering the `fetchExperimentAssignment` function, if it throws an error we return the default experience:

```js
export const fetchExperimentAssignment = async ( {
	experimentName,
	anonId,
}: {
	experimentName: string;
	anonId: string | null;
} ): Promise< unknown > => {

if (!window.wcTracks.isEnabled) {
    throw new Error('Tracking is disabled')
}
```

Reply:
I see this working. The majority of users won't have the data sharing turned on, so we'd need to be ok with this error being thrown for most people.

When an error is shown, is the Experiment thrown out for that user or do they fall into the default experience bucket? If so, I'm wondering if that may skew the data?

Reply:
> When an error is shown, is the Experiment thrown out for that user or do they fall into the default experience bucket?

Throwing this error causes:
- No assignment to occur - they won't get included in the experiment analysis.
- The default experience to show - the user won't notice anything.

> I'm wondering if that may skew the data?

Good point!

There are two factors to consider here:
1. Internal Validity - the correctness of the results without attempting to generalise them to other populations.
2. External Validity - the extent to which results can be generalised to other populations.

Experimenting like this should be internally valid as whether or not a user turns on data sharing is independent of which variation they are assigned to.

However there will be limits to external validity, the population we are testing will be the population of experimenters that enable tracking, the extent to which this would generalise to the full population is hard to say and will likely depend on the nature of the experiment. There is the same issue with using tracks stats to determine Woo's development and I feel it is a matter of "This is the only data can get." as well as keeping in mind that this limitation exists.

I might get @yanirs to confirm that my understanding is correct.

Reply:
:+1: that's a good explanation @jessie-ross.

Reply:
Ah, I think it would be great to document this in the readme

Reply:
Thanks for the explanation, it makes lots of sense. 

I think you've already answered and its probably a non issue, but just to clarify, I was asking about the mechanism of what happens to a user whose function call `fetchExperimentAssignment` throws an error. Is that user thrown out of the experiment and shown the default experience or is that user assigned to the default group and shown the default experience?

I just want to make sure it's not the latter, as that would cause problems because the user will be unable to send Tracks events which may be used to determine the effectiveness of the experiment.

The next question is: Is there a way to not show the error and/or log it in production? The majority of Woo users have opted out of data sharing, so there will be a lot of noise in LogStash and these users will be getting many errors thrown in the console. Can we avoid this?

Reply:
>  Is that user thrown out of the experiment and shown the default experience or is that user assigned to the default group and shown the default experience?

The user will be thrown out and shown the default experience - they won't affect the experiment results.

> The next question is: Is there a way to not show the error and/or log it in production? The majority of Woo users have opted out of data sharing, so there will be a lot of noise in LogStash and these users will be getting many errors thrown in the console. Can we avoid this?

If you include [this change](https://github.com/woocommerce/woocommerce-admin/pull/6669#discussion_r604715238) as well, users that have tracking disabled also won't send any log stash errors. This means users that are thrown out will effectively be invisible.

Reply:
Thanks for the response and confirmation. I like the simplicity of this approach, but am hesitant to throw Errors when there isn't an error and the code is working as intended. Would an early return from `fetchExperimentAssignment` have the same effect as throwing an Error?

Reply:
>but am hesitant to throw Errors when there isn't an error and the code is working as intended

I just tried a production build and those errors are caught and handled without being surfaced to the console, so I'm on board with this approach